cmd/pmm-configurator/sshkey.go

```diff
@@ -3,109 +3,28 @@ package main
 import (
 	"encoding/json"
 	"fmt"
-	"io/ioutil"
 	"net/http"
-	"os"
-	"os/user"
-	"path"
-	"path/filepath"
-	"strconv"
 
-	log "github.com/sirupsen/logrus"
-	"golang.org/x/crypto/ssh"
-	"golang.org/x/sys/unix"
+	"github.com/percona/pmm-manage/configurator/sshkey"
 )
 
-func runSSHKeyChecks() {
-	sshKeyUser, err := user.Lookup(c.SSHKeyOwner)
-	if err != nil {
-		log.Fatal(err)
-	}
-	if c.SSHKeyPath == "" {
-		c.SSHKeyPath = path.Join(sshKeyUser.HomeDir, ".ssh/authorized_keys")
-	}
-
-	sshKeyDir := filepath.Dir(c.SSHKeyPath)
-	if dir, err := os.Stat(sshKeyDir); err != nil || !dir.IsDir() {
-		if err := os.MkdirAll(sshKeyDir, 0700); err != nil {
-			log.WithFields(log.Fields{
-				"dir":   sshKeyDir,
-				"error": err,
-			}).Fatal("Cannot create ssh directory")
-		}
-		uid, _ := strconv.Atoi(sshKeyUser.Uid)
-		gid, _ := strconv.Atoi(sshKeyUser.Gid)
-		if err := os.Chown(sshKeyDir, uid, gid); err != nil {
-			log.WithFields(log.Fields{
-				"dir":   sshKeyDir,
-				"error": err,
-			}).Fatal("Cannot change owner of ssh directory")
-		}
-	}
-	if err := unix.Access(sshKeyDir, unix.W_OK); err != nil {
-		log.WithFields(log.Fields{
-			"dir":   sshKeyDir,
-			"error": err,
-		}).Fatal("Cannot write to ssh directory")
-	}
-}
-
-func parseSSHKey(authorizedKey []byte) (sshkey, error) {
-	pubKey, comment, _, _, err := ssh.ParseAuthorizedKey(authorizedKey)
-	if err != nil {
-		return sshkey{}, err
-	}
-	return sshkey{
-		Type:        pubKey.Type(),
-		Comment:     comment,
-		Fingerprint: ssh.FingerprintSHA256(pubKey),
-	}, err
-}
-
 func getSSHKeyHandler(w http.ResponseWriter, req *http.Request) {
-	authorizedKey, err := ioutil.ReadFile(c.SSHKeyPath)
-	if err != nil {
-		returnError(w, req, http.StatusInternalServerError, "Cannot read ssh key", err)
-		return
+	parsedSSHKey, result, err := sshkey.ReadSSHKey()
+	if result != "success" {
+		returnError(w, req, http.StatusInternalServerError, result, err)
+	} else {
+		json.NewEncoder(w).Encode(parsedSSHKey) // nolint: errcheck
```

Comment:
cognitive load is fixed https://github.com/percona/pmm-manage/pull/9/commits/d2a51dfe001307aa02fd0f1ec4e305ea8219a262 :)